osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -49,28 +52,33 @@ public BeatmapSetInfo BeatmapSet
         }
 
         private void updateDisplay()
+        {
+            updateChildren();
+
+            Beatmap.Value = BeatmapSet?.Beatmaps.FirstOrDefault();
+            plays.Value = BeatmapSet?.OnlineInfo.PlayCount ?? 0;
+            favourites.Value = BeatmapSet?.OnlineInfo.FavouriteCount ?? 0;
+        }
+
+        private void updateChildren()
         {
             difficulties.Clear();
 
-            if (BeatmapSet != null)
+            if (BeatmapSet == null)
+                return;
+
+            difficulties.ChildrenEnumerable = BeatmapSet.Beatmaps.OrderBy(beatmap => beatmap.StarDifficulty).ToList().FindAll(b => b.Ruleset.ID == Ruleset.Value.ID).Select(b => new DifficultySelectorButton(b)
```

Comment:
Should probably use `RulesetID` instead,

Reply:
afaik, `RulesetID` is a database thing which is used locally, `Ruleset.ID` is what should be used for online-fetched info